File: tests/algorithms/x/Java/maths/sieve_of_eratosthenes.java
public class Main {

    static int isqrt(int n) {
        int r = 0;
        while ((r + 1) * (r + 1) <= n) {
            r = r + 1;
        }
        return r;
    }

    static int[] prime_sieve(int num) {
        if (num <= 0) {
            throw new RuntimeException(String.valueOf("Invalid input, please enter a positive integer."));
        }
        boolean[] sieve = ((boolean[])(new boolean[]{}));
        int i = 0;
        while (i <= num) {
            sieve = ((boolean[])(appendBool(sieve, true)));
            i = i + 1;
        }
        int[] prime = ((int[])(new int[]{}));
        int start = 2;
        int end = isqrt(num);
        while (start <= end) {
            if (((Boolean)(sieve[start]))) {
                prime = ((int[])(java.util.stream.IntStream.concat(java.util.Arrays.stream(prime), java.util.stream.IntStream.of(start)).toArray()));
                int j = start * start;
                while (j <= num) {
                    if (((Boolean)(sieve[j]))) {
sieve[j] = false;
                    }
                    j = j + start;
                }
            }
            start = start + 1;
        }
        int k = end + 1;
        while (k <= num) {
            if (((Boolean)(sieve[k]))) {
                prime = ((int[])(java.util.stream.IntStream.concat(java.util.Arrays.stream(prime), java.util.stream.IntStream.of(k)).toArray()));
            }
            k = k + 1;
        }
        return prime;
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            System.out.println(_p(prime_sieve(50)));
            System.out.println(_p(prime_sieve(25)));
            System.out.println(_p(prime_sieve(10)));
            System.out.println(_p(prime_sieve(9)));
            System.out.println(_p(prime_sieve(2)));
            System.out.println(_p(prime_sieve(1)));
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static boolean[] appendBool(boolean[] arr, boolean v) {
        boolean[] out = java.util.Arrays.copyOf(arr, arr.length + 1);
        out[arr.length] = v;
        return out;
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/transpiler/FS/maths/sieve_of_eratosthenes.fs
// Generated 2025-08-08 18:58 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _dictAdd<'K,'V when 'K : equality> (d:System.Collections.Generic.IDictionary<'K,'V>) (k:'K) (v:'V) =
    d.[k] <- v
    d
let _dictCreate<'K,'V when 'K : equality> (pairs:('K * 'V) list) : System.Collections.Generic.IDictionary<'K,'V> =
    let d = System.Collections.Generic.Dictionary<'K, 'V>()
    for (k, v) in pairs do
        d.[k] <- v
    upcast d
let _dictGet<'K,'V when 'K : equality> (d:System.Collections.Generic.IDictionary<'K,'V>) (k:'K) : 'V =
    match d.TryGetValue(k) with
    | true, v -> v
    | _ -> Unchecked.defaultof<'V>
let _idx (arr:'a array) (i:int) : 'a =
    if not (obj.ReferenceEquals(arr, null)) && i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
let _arrset (arr:'a array) (i:int) (v:'a) : 'a array =
    let mutable a = arr
    if i >= a.Length then
        let na = Array.zeroCreate<'a> (i + 1)
        Array.blit a 0 na 0 a.Length
        a <- na
    a.[i] <- v
    a
let rec _str v =
    let s = sprintf "%A" v
    let s = if s.EndsWith(".0") then s.Substring(0, s.Length - 2) else s
    s.Replace("[|", "[")
     .Replace("|]", "]")
     .Replace("; ", " ")
     .Replace(";", "")
     .Replace("\"", "")
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec isqrt (n: int) =
    let mutable __ret : int = Unchecked.defaultof<int>
    let mutable n = n
    try
        let mutable r: int = 0
        while ((int64 (r + 1)) * (int64 (r + 1))) <= (int64 n) do
            r <- r + 1
        __ret <- r
        raise Return
        __ret
    with
        | Return -> __ret
let rec prime_sieve (num: int) =
    let mutable __ret : int array = Unchecked.defaultof<int array>
    let mutable num = num
    try
        if num <= 0 then
            failwith ("Invalid input, please enter a positive integer.")
        let mutable sieve: bool array = Array.empty<bool>
        let mutable i: int = 0
        while i <= num do
            sieve <- Array.append sieve [|true|]
            i <- i + 1
        let mutable prime: int array = Array.empty<int>
        let mutable start: int = 2
        let ``end``: int = isqrt (num)
        while start <= ``end`` do
            if _idx sieve (int start) then
                prime <- Array.append prime [|start|]
                let mutable j: int64 = (int64 start) * (int64 start)
                while j <= (int64 num) do
                    if _idx sieve (int j) then
                        sieve.[int j] <- false
                    j <- j + (int64 start)
            start <- start + 1
        let mutable k: int = ``end`` + 1
        while k <= num do
            if _idx sieve (int k) then
                prime <- Array.append prime [|k|]
            k <- k + 1
        __ret <- prime
        raise Return
        __ret
    with
        | Return -> __ret
printfn "%s" (_str (prime_sieve (50)))
printfn "%s" (_str (prime_sieve (25)))
printfn "%s" (_str (prime_sieve (10)))
printfn "%s" (_str (prime_sieve (9)))
printfn "%s" (_str (prime_sieve (2)))
printfn "%s" (_str (prime_sieve (1)))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
